walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -27,35 +27,41 @@
   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
 
-    ArrayList<TimeRange> allUnavailableTimes = new ArrayList<>();
+    ArrayList<TimeRange> mandatoryUnavailableTimes = new ArrayList<>();
+    ArrayList<TimeRange> optionalUnavailableTimes = new ArrayList<>();
 
     for (Event event : events){
-      if (!isEventBlockingRequest(event, request)){
-        allUnavailableTimes.add(event.getWhen());
-      }
-    }
 
-    ArrayList<TimeRange> mergedUnavailableTimes = mergeTimeRanges(allUnavailableTimes);
-    ArrayList<TimeRange> availableTimes = new ArrayList<>();
-    
-    TimeRange possibleTime;
-    int prevEndTime = TimeRange.START_OF_DAY;
-    
-    for (TimeRange curTime: mergedUnavailableTimes){
-
-      possibleTime = TimeRange.fromStartEnd(prevEndTime, curTime.start(), false);
+      if (!isEventBlockingRequest(event, request, true)){
+        mandatoryUnavailableTimes.add(event.getWhen());
+      }
 
-      if (possibleTime.duration() >= request.getDuration()){
-        availableTimes.add(possibleTime);
+      if(!isEventBlockingRequest(event, request, false)){
+        optionalUnavailableTimes.add(event.getWhen());
       }
+    }
 
-      prevEndTime = curTime.end();
+    ArrayList<TimeRange> unavailableTimes;
+    boolean combinedUnavailableTimes = false;
```

Comment:
Just something I noticed here-- these 2 variable names got me a little turned around at first glance since they're so similar but completely different types. This might just be personal preference, but using something like "allUnavailableAdded" might be a little clearer here-- to me, combinedUnavailableTimes sounded like it was supposed to be an ArrayList with all of the times (again this is a super minor nit and in the end probably just up to what you prefer)

Reply:
Thanks for pointing that out! I changed the naming so it'd be more clearer.